client/my-sites/controller.js

```diff
@@ -338,41 +369,45 @@ module.exports = {
 	},
 
 	sites( context ) {
-		const SitesComponent = require( 'my-sites/sites' );
-		const analyticsPageTitle = 'Sites';
-		const basePath = route.sectionify( context.path );
-		const path = context.prevPath ? route.sectionify( context.prevPath ) : '/stats';
-
 		if ( context.query.verified === '1' ) {
-			notices.success( i18n.translate( "Email verified! Now that you've confirmed your email address you can publish posts on your blog." ) );
+			notices.success(
+				i18n.translate(
+					"Email verified! Now that you've confirmed your email address you can publish posts on your blog."
+				)
+			);
 		}
 		/**
 		 * Sites is rendered on #primary but it doesn't expect a sidebar to exist
 		 */
 		removeSidebar( context );
 		context.store.dispatch( setLayoutFocus( 'content' ) );
 
-		// This path sets the URL to be visited once a site is selected
-		const sourcePath = ( basePath === '/sites' ) ? path : basePath;
-
-		analytics.pageView.record( basePath, analyticsPageTitle );
-
 		renderWithReduxStore(
-			React.createElement( SitesComponent, {
-				sites,
-				path: context.path,
-				sourcePath,
-				user,
-				getSiteSelectionHeaderText: context.getSiteSelectionHeaderText,
-				trackScrollPage: trackScrollPage.bind(
-					null,
-					basePath,
-					analyticsPageTitle,
-					'Sites'
-				)
-			} ),
+			createSitesComponent( context ),
 			document.getElementById( 'primary' ),
 			context.store
 		);
-	}
+	},
+
+	/**
+	 * Middleware that adds the site selector screen to the layout
+	 * without rendering the layout. For use with isomorphic routing
+	 * @see {@link https://github.com/Automattic/wp-calypso/blob/master/docs/isomorphic-routing.md }
+	 *
+	 * To show the site selector screen using traditional multi-tree
+	 * layout, use the sites() middleware above.
+	 *
+	 * @param {object} context -- Middleware context
+	 * @param {function} next -- Call next middleware in chain
+	 */
+	sitesWithoutRender( context, next ) {
```

Comment:
My nomenclature was to prefix with `make`, i.e. `makeSites`, in this case. Cf [`makeNavigation`](https://github.com/Automattic/wp-calypso/blob/2725025cf4a3d808291acb3d312910b337f937f6/client/my-sites/controller.js#L303).

(Yay for the JSDoc!)